packages/node-provider/src/node-provider.ts

```diff
@@ -1,7 +1,35 @@
-import { INodeProvider, NodeMessage } from "./types";
+import {
+  INodeProvider,
+  NodeMessage,
+  NodeMessageReceivedCallback,
+  NodeMessageType,
+  NodeProviderStatus
+} from "./types";
 
 export default class NodeProvider implements INodeProvider {
-  public onMessage(callback: (message: NodeMessage) => void) {}
+  private internalStatus: NodeProviderStatus = NodeProviderStatus.OFFLINE;
```

Comment:
Since we require a `connect` method to bridge the Playground to the iframe, I thought this would be a good idea to prevent unwanted operations until the NodeProvider is fully connected. Thoughts?

Reply:
Seems fine

Reply:
I think I like `isConnected` better

Reply:
This `connect` method is only a temporary solution until the node runs in an extension/mobile app, right? The original idea was that a fully-initialized NodeProvider would be injected into the page.

Reply:
@mvanderh I’m not sure it should be temporary. I am not particularly fond of the idea of injecting a global variable into every webpage, but that can be a debate for another day

Reply:
For comparison, Metamask and other dApp wallets inject a global `ethereum` object into the browser context.  But yes, let's talk another day.

Reply:
Yeah I’m familiar with that and think it’s not great